easybuild/tools/modules.py

```diff
@@ -543,9 +546,25 @@ def mod_exists_via_show(mod_name):
 
             :param mod_name: module name
             """
-            mod_exists_regex = mod_exists_regex_template % re.escape(mod_name)
-            txt = self.show(mod_name)
-            return bool(re.search(mod_exists_regex, txt, re.M))
+            stderr = self.show(mod_name)
+            res = False
+            # Parse the output:
+            # - Skip whitespace
+            # - Any error -> Module does not exist
+            # - Check first non-whitespace line for something that looks like an absolute path terminated by a colon
+            mod_exists_regex = r'\s*/.+:\s*'
```

Comment:
Hmm, this quite liberal now... No reference to part of `mod_name` requires at all anymore?

Reply:
No. As the test with the alias shows: You can alias `foo123`->`bar/4.5.6` if you wish to do so.

As all tests (including the negative tests) show that this works, it is fine.

Stronger argument: LMods stderr is empty if the module does not exist. EnvironmentModules output contains the "ERROR" first (and only) which is also caught. Furthermore only the first non-comment line is checked and the patter is `<space><absolute-path>:<space>` which is quite restrictive

Reply:
OK, the alias thing prevents us from making this more restrictive, indeed.

But maybe we should check whether the path for which the patterns matches actually exists? Not sure if that would take us too far...

Reply:
Yeah, ok, it's probably overkill indeed...